pkg/pthi/commands.go

```diff
@@ -39,6 +40,61 @@ func NewCommand() Command {
 	}
 }
 
+func GetOSIPAddress(results []uint8) (uint32, error) {
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return 0, errors.New("fail")
+	}
+
+	zero_bytes := []byte{0, 0, 0, 0, 0, 0, 0}
+
+	if bytes.Equal([]byte(results), zero_bytes) {
+		return 0, nil
+	}
+
+	for _, iface := range interfaces {
+		if iface.Flags&net.FlagUp == 0 {
+			continue // interface down
+		}
+		if iface.Flags&net.FlagLoopback != 0 {
+			continue // loopback interface
+		}
+
+		hwaddr := iface.HardwareAddr
+
+		if bytes.Equal(hwaddr, []byte(results)) {
+			addrs, err := iface.Addrs()
+			if err != nil {
+				return 0, errors.New("fail")
```

Comment:
ideally, a bit more descriptive error message would be good. For example "failed to get interface addresses"